yt_idefix/data_structures.py

```diff
@@ -724,3 +797,241 @@ def _validate_units_override_keys(cls, units_override):
             )
 
         super(cls, cls)._validate_units_override_keys(units_override)
+
+
+class PlutoXdmfDataset(PlutoVtkDataset):
+    _index_class = PlutoXdmfHierarchy
+    _field_info_class = PlutoXdmfFields
+    _dataset_type = "pluto-xdmf"
+    _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
+    _required_header_keyword = "PLUTOXdmf"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
+        grid_file = os.path.join(os.path.dirname(self.parameter_filename), "grid.out")
+        self.parameter_filename[:-2] + "xmf"
+        out_file = os.path.join(
+            os.path.dirname(self.parameter_filename),
+            self.parameter_filename[-6:] + ".out",
+        )
+        self._default_inifile = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_inifile
+        )
+        self._default_definitions_header = os.path.join(
+            os.path.dirname(self.parameter_filename), self._default_definitions_header
+        )
+        with open(grid_file) as gridtxt:
+            txt = gridtxt.readlines()
+            self.domain_left_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[2]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.domain_right_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[3]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.dimensionality = int(txt[4][-2])
+            self.domain_dimensions = np.array(
+                [
+                    int(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[4]
+                    )
+                    for i in [6, 7, 8]
+                ][: self.dimensionality]
+            )
+
+            def parse_geometry(geom: str):
+                import yt
+
+                if yt.version_info[:2] > (4, 1):
+                    try:
+                        from yt.geometry.api import (
+                            Geometry,  # type: ignore [attr-defined]
+                        )
+
+                        return Geometry(geom)
+                    except ImportError:
+                        pass
+
+                return geom
```

Comment:
this definition shouldn't be nested in the `with` block. Also it's duplicated from `IdefixDataset` and I don't think it's necessary.
Because this class inherits from `IdefixDataset` I would expect the geometry parsing to happen via the parent class, with `super()._parse_parameter_file()` (or `IdefixDataset._parse_parameter_file(self)` if you need to skip `IdefixVtkDataset`)
Please tell me if anything doesn't work with my proposed approach. The intention is to save a bunch of repetitions and simplify your own specialised class

Reply:
There were other stuffs like reading vtk that was done by IdefixVtkDataset._parse_parameter_file(self). So I copied this part of the code. Maybe using IdefixDataset would fix this.

Reply:
I want to use only the ```parse_geometry``` function while not anything else in ```_setup_geometry```. Is there a way to acomplish this taking advantage of the inheritance and avoiding repetitions? This is because the rest of the code in ```_setup_geometry``` is not suited for the xdmf files and causes errors if called. @neutrinoceros  Can you suggest a solution?

Reply:
I did a small change in the code in how parse geometry was placed. Let me know if this is okay.